xonsh/aliases.py

```diff
@@ -402,6 +402,8 @@ def which(args, stdin=None, stdout=None, stderr=None):
                     print(arg, file=stdout)
             else:
                 print("aliases['{}'] = {}".format(arg, builtins.aliases[arg]), file=stdout)
+                if not isinstance(builtins.aliases[arg], list):
```

Comment:
That one is probably wrong too.  It should at least be an `abc.Sequence` test above.  And this one is explicitly for callables